File: Programmers/Lv1/Lv1_체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    int* uniform = new int[n + 1];
    fill(uniform, uniform + n + 1, 0);
    for (int i = 0; i < reserve.size(); i++) {
        uniform[reserve[i]]++;
    }
    for (int i = 0; i < lost.size(); i++) {
        uniform[lost[i]]--;
    }
    for (int i = 1; i < n + 1; i++) {
        if (uniform[i] == -1) {
            if (i > 1 && uniform[i - 1] == 1) {
                uniform[i - 1]--;
                uniform[i] = 0;
            }
            else if (i < n && uniform[i + 1] == 1) {
                uniform[i + 1]--;
                uniform[i] = 0;
            }
        }
        if (uniform[i] >= 0) {
            answer++;
        }
    }
    return answer;
}

File: Programmers/Lv1/체육복/Lv1_체육복.py
def solution(n, lost, reserve):
    answer = 0
    uniform = [0 for i in range(n)]
    for i in lost:
        uniform[i-1] -= 1
    for i in reserve:
        uniform[i-1] += 1
    for i in range(n):
        if uniform[i] == -1:
            if i > 0 and uniform[i-1] > 0:
                uniform[i-1] -= 1
                uniform[i] += 1
            elif i < n-1 and uniform[i+1] > 0:
                uniform[i+1] -= 1
                uniform[i] += 1
        if uniform[i] > -1:
            answer += 1
    return answer